arch/sim/src/sim/up_oneshot.c

```diff
@@ -299,6 +300,67 @@ static int sim_current(FAR struct oneshot_lowerhalf_s *lower,
   return OK;
 }
 
+#ifdef CONFIG_SIM_ONESHOT_HOST_TIMER
+/****************************************************************************
+ * Name: host_signal_handler
+ *
+ * Description:
+ *   The signal handler is called periodically and is used to deliver TICK
+ *   events to the OS.
+ *
+ * Input Parameters:
+ *   sig - the signal number
+ *   si  - the signal information
+ *   old_ucontext - the previous context
+ *
+ ****************************************************************************/
+
+static int host_sigalrm_handler(int irq, FAR void *context, FAR void *arg)
+{
+  up_timer_update();
+  return OK;
+}
+
+/****************************************************************************
+ * Name: host_init_timer
+ *
+ * Description:
+ *   Creates a periodic timer and sets up a signal handler on the host.
+ *
+ * Input Parameters:
+ *   nxsched_process_timer - the NuttX scheduler callback
+ *
+ * Returned Value:
+ *   This function returns 0 on success otherwise a negative error code.
+ *
+ ****************************************************************************/
+
+static int host_init_timer(void)
```

Comment:
Merge the code into up_timer_initialize?

Reply:
I merged it